WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -20,6 +25,12 @@ final class AuthenticatedImageDownload: AsyncOperation {
     override func main() {
         let result = ContextManager.shared.performQuery { context in
             Result {
+                // Can't fetch the blog using a temporary ID. This check is added as an attempt to prevent this crash:
+                // https://github.com/wordpress-mobile/WordPress-iOS/issues/20630
+                guard !self.blogObjectID.isTemporaryID else {
+                    throw DownloadError.blogNotFound
```

Comment:
Have you considered making this error more specific about the blog being a temporary ID?

Reply:
I used a quite generic "blog not found" error, because I thought the caller won't care about this detail—whether it's caused by an invalid object id or the blog that associated with the object id is deleted. However, that's all hypothetical though, because no one handles this error. The error type is added just to make the compiler happy 🥲

Reply:
> The error type is added just to make the compiler happy 🥲

Excellent emoji choice.